libsolidity/analysis/TypeChecker.cpp

```diff
@@ -112,8 +112,9 @@ void TypeChecker::checkDoubleStorageAssignment(Assignment const& _assignment)
 	size_t storageByteAccesses = 0;
 	auto count = [&](TupleExpression const& _lhs, TupleType const& _rhs, auto _recurse) -> void {
 		TupleType const& lhsType = dynamic_cast<TupleType const&>(*type(_lhs));
+		TupleExpression const* lhs = dynamic_cast<TupleExpression const*>(resolveOuterUnaryTuples(&_lhs));
```

Comment:
I think this var could have a more distinguishable name from the parameter var. What about `lhsResolved` or something like that?

Reply:
Done